templates/definition/meter/sungrow-hybrid.yaml

```diff
@@ -224,6 +228,33 @@ render: |
               address: 13050 # Charge/discharge command
               type: writesingle
               decode: uint16
+        {{- if ne .maxchargepower "" }}
+        - source: go
+          script: power
+          in:
+            - name: power
+              type: int
+              config:
+                source: calc
+                mul:
+                - source: const
+                  value: {{ .maxchargerate }}
+                - source: const
+                  value: {{ .maxchargepower }}
+                - source: const
+                  value: 0.01
+          out:
+          - name: power
+            type: int
+            config:
+              source: modbus
+              {{- include "modbus" . | indent 12 }}
+              timeout: {{ .timeout }}
+              register:
+                address: 13051 # Charge/discharge power
+                type: writesingle
+                decode: uint16
```

Comment:
Falls wirklich rate+power notwendig sind kann die Multiplikation einfach im Template passieren:

    {{ .maxchargerate * .maxchargepower }}

...das sollte dann ohne Go gehen.

Reply:
erstmal Bedarf klären!

Reply:
> Falls wirklich rate+power notwendig sind kann die Multiplikation einfach im Template passieren (ungetestet):
> 
> ```
> {{ .maxchargerate * .maxchargepower }}
> ```
> 
> ...das sollte dann ohne Go gehen.

Mit "*" kommt folgender Fehler 
```
cannot create meter 'grid1': cannot create meter type 'template': template: eebus.tpl:205: unexpected "*" in operand
```
mit mul klappt es
```
        {{- if ne .maxchargepower "" }}
        - source: const
          value: {{ mul .maxchargerate .maxchargepower 0.01 }}  # maxchargerate * maxchargepower * 0,01 
          set:
            source: modbus
            {{- include "modbus" . | indent 10 }}
            timeout: {{ .timeout }}
            register:
              address: 13051 # Charge/discharge power
              type: writesingle
              decode: uint16
        {{- else }}
```

Reply:
> erstmal Bedarf klären!

Es würde "maxchargerate" ausreichen um die Ladeleistung vom Wechselrichter zu begrenzen.

Für "maxchargepower" spricht, wenn die Batterie weniger kann als der Wechselrichter. 
z.B.
SBR096 DC Nennleistung wäre 5.76KW (Battery voltage * maximaler Ladestrom des Wechselrichters (SH*RT: 30A)/SH*T:50A)
SH8.0RT wäre 8KW Ladeleistung möglich. (BDC rated power * 100)

Dies kann aber dann wieder zu Fehlern bei den Anwendern führen, wenn ein höherer Ladestrom an der Batterie möglich ist, als der Wechselrichter in der Lage ist zu liefern.
z.B. SBR160 (9,60 kW) mit SH8.0RT (8KW)